### ISSUE TITLE
[Question] Embedded async relationship with Symmetric Multiprocessing (SMP)?

### ISSUE THREAD

**Issue by awsa2ron** [embassy-rs/embassy#2365]
It's a open question. 
I found this [Symmetric Multiprocessing (SMP) with FreeRTOS ](https://www.freertos.org/symmetric-multiprocessing-introduction.html)

Just wondering deeper insight form you.

**Comment by lights0123** [comment]
SMP is totally supported—one of embassy's supported chips, the RP2040, operates in SMP with dual cores. Here's an example of activating the second core with embassy: https://github.com/embassy-rs/embassy/blob/main/examples/rp/src/bin/multicore.rs

**Comment by lights0123** [comment]
I don't believe Embassy has any sort of multicore scheduler though—you spawn tasks on a specific core and they will never switch, so tasks on the same core will never run in parallel. Tasks running on different cores will run in parallel, though.

**Comment by enbyted** [comment]
Small clarification on the terms used. I think embassy does not support **SMP** (S = Symmetric), however it supports **AMP** (A = Asymmetric).
The difference being that in SMP any processing unit takes work from a shared queue (as is the case in FreeRTOS SMP), while in AMP each processing unit has a predefined list of tasks that it may execute.

I guess that you could implement something like having a few executors that are shared between cores, that you then `.pool()` whichever executor has work to do from whichever core is free, which would achieve SMP, but it's not build-in, I don't think.
